core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -5183,12 +5193,12 @@ else if (!(isEffectivelyConstantExpression(node) && n > 1)) {
             }
         }
 
+        // check if innerVirtualModel is trivial, e.g, it does not contain any arithmetic
         if (useInnerModel) {
             final ObjList<QueryColumn> innerColumns = innerVirtualModel.getBottomUpColumns();
             useInnerModel = false;
             boolean columnsAndFunctionsOnly = true;
             // hour(column) is the only function key in supported by Rosti, so we need to detect it
```

Comment:
This comment should be removed now.

Reply:
eh? why, i added this comment on purpose. This is the sole objective of the code inside the  "if" statement

Reply:
sorry, i misunderstood which comment you were referring to, removed here: https://github.com/questdb/questdb/pull/4690/commits/3eb393ad8c6ca229e87feba3128b3835b5b97905